### PR TITLE
Update shell configuration, test pnpm setup, simplify

### PR DESCRIPTION
**Pull request by ProchaLu**
We have multiple steps for setting up `pnpm`, and sometimes students miss running a command that `pnpm` shows after running `pnpm setup`.

```bash
To start using pnpm, run:
source /Users/lukas/.zshrc
```

If students don’t run this source command, `pnpm` will fail when using `pnpm add`:

```bash
➜  ~ pnpm add --global @upleveled/preflight
 ERR_PNPM_NO_GLOBAL_BIN_DIR  Unable to find the global bin directory

Run "pnpm setup" to create it automatically, or set the global-bin-dir setting, or the PNPM_HOME env variable. The global bin directory should be in the PATH.
```

To avoid this error, this PR updates the setup instructions to make sure students run the necessary source command, preventing issues with `pnpm`.

- [x] Add `source` commands after `pnpm setup`
- [x] Add test that the `source` commands worked (move installation of `@upleveled/preflight` further up)
- [x] Remove `$USERPROFILE` from Windows guide, reorder steps if necessary

Additionally, this PR also removes the section for running `pnpm create @upleveled/react-app --help`, as this only displays the help information and we show in the first front-end React lecture how to set up `@upleveled/create-react-app`.

### Additional Information:

`.bashrc` is read only once, when bash starts. If you make any subsequent changes to `.bashrc`, they won’t be applied until .bashrc is re-read. By running `source .bashrc`, you are telling bash to re-read that file.

Alternatively, you can start a new bash process (e.g., by opening a new terminal session), which will cause the new bash process to read the .bashrc file again. However, this will not affect the old session.

### CODE REVIEW

**Comment by karlhorky** [comment]
Hm, on Windows, we now alternate between `$USERPROFILE` (step 7, step 19), `$HOME` (step 23) and `~` (step 23)

Let's keep this to only `$HOME` and `~`:

1. Replace `$USERPROFILE` in the `echo` commands with `~` (I think we can probably also remove the double quotes)
2. Anywhere we're running `source`, let's use `~` on Windows (also double quotes can go)

The problem with changing the `source` on step 7 to use `~` is that [it appears PowerShell doesn't support tilde](https://www.reddit.com/r/PowerShell/comments/14wdeih/tilde/)

We could move the Hyper install + configuration with Git Bash up earlier (eg. as a part of step 6 or a new step between step 6 and 7)

**Comment by ProchaLu** [comment]
## Windows 

I tested the System Setup Guide for installing Node.js and pnpm using Corepack on Windows. I already had these tools installed, I first removed them by running `choco uninstall nodejs-lts` and deleting all directories named `node` and `pnpm`. To confirm the uninstallation, I used the command `node -v`, which returned the following error:

```bash
The term 'node' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```

This showed that Node.js was removed from the system. I reinstalled Node.js using the command `choco install nodejs-lts`. After installation, Node.js was on the system, but pnpm was not. I followed the remaining System Setup steps, and the installation of pnpm completed without any issues.

node

File: windows.md [hunk 1]
```diff
@@ -54,22 +54,36 @@ With those compatibility things out of the way, you're ready to start the system
    ```bash
    choco install slack -y
    ```
-7. Copy the following text and right-click in the blue middle part of the PowerShell window to paste the text. Hit enter.<br><br>
+7. Close PowerShell and open it again as administrator (like in step 2). Copy the following text and right-click in the blue middle part of the PowerShell window to paste the text. Hit enter.<br><br>
```
Comment (ProchaLu) [inline windows.md:57]: We need to reopen the terminal because, after installing with choco, the environment variables changed, as explained in this [comment](https://github.com/upleveled/system-setup/pull/91#issuecomment-2270597504).

Comment (karlhorky) [reply windows.md:57]: [the linked comment](https://github.com/upleveled/system-setup/pull/91#issuecomment-2270597504) doesn't explain a reason why we should move the "close and reopen"  from step 8 to step 7 in the Windows guide

the comment explains that `node -v` didn't work, but that's not a command that we run in step 7

check into this again - if we actually have a problem with `corepack` being unavailable, then of course we can make this or other changes. but I'm guessing that we don't have an issue here.

File: linux.md [hunk 2]
```diff
@@ -28,11 +28,23 @@ Make sure that you're running the 2nd-newest OS version or the newest version -
...
+   pnpm add --global @upleveled/preflight
+   ```
+   This will prepare a program that we will use in the course. If you encounter the following error: 
+   ```bash
+   ERR_PNPM_NO_GLOBAL_BIN_DIR  Unable to find the global bin directory
+
+   Run "pnpm setup" to create it automatically, or set the global-bin-dir setting, or the PNPM_HOME env variable. The global bin directory should be in the PATH.
+   ```
+   This indicates that the `pnpm` installation was incomplete- check the last part of the previous step to see how to finish the installation.<br><br>
+5. Copy each line in the following text, paste it in the terminal and hit return.<br><br>
```
Comment (karlhorky) [inline linux.md:47]: we can merge this into the previous step, to indicate that the previous step is not "done" yet:

````

   This uses Corepack to install `pnpm`, and configures `pnpm`'s global bin directory.<br><br>

   Install `@upleveled/preflight`, a program we will use in the course, to verify that the previous commands were successful: copy the following text, paste it in the terminal and hit return.<br><br>

   ```bash
   pnpm add --global @upleveled/preflight
   ```

   Verify that your output looks similar to the "Successful" output below. If it's more like the "Failing" output, repeat the pnpm installation commands at the start of this step.

   <table>
      <tr>
         <th>Successful</th>
         <th>Failing</th>
      </tr>
      <tr>
         <td valign="top">

   ```bash
    WARN  6 deprecated subdependencies found:
   @oclif/screen@3.0.8, glob@6.0.4, glob@7.1.6,
   osenv@0.1.5, rimraf@2.4.5, rimraf@3.0.2
   Packages: +21 -32
   +++++++++++++++++--------------------------
   Progress: resolved 773, reused 772,
   downloaded 1, added 21, done
   
   /Users/k/Library/pnpm/global/5:
   + @upleveled/preflight 7.0.8
   
   Done in 3.3s
   ```

   </td>
   <td valign="top">

   ```bash
   ERR_PNPM_NO_GLOBAL_BIN_DIR  Unable to find the global
   bin directory
   
   Run "pnpm setup" to create it automatically, or set the
   global-bin-dir setting, or the PNPM_HOME env variable.
   The global bin directory should be in the PATH.
   ```

   </td></tr></table>

5. Copy each line in the following text, paste it in the terminal and hit return.<br><br>
````

the numbering, starting with 5 at the end of the code block, needs to be updated to reflect the removed step (5 becomes 4, 6 becomes 5, etc)

apply this to all of the guides

this can be the styling for all of the steps in future, so that we have built-in verification of all steps

rendered looks like this:

Screenshot 2024-08-13 at 14 46 35

File: linux.md [hunk 3]
```diff
@@ -101,25 +113,11 @@ Make sure that you're running the 2nd-newest OS version or the newest version -
...
-
-    This will prepare a program that we will use in the course. This will take a while and then respond with a message with some usage instructions, similar to this:<br><br>
-    <img src="./general-cra-installed.avif"><br><br>
-
-14. Copy the following text, paste it in the terminal and hit return.<br><br>
-    ```bash
-    pnpm add --global @upleveled/preflight
-    ```
-    This will prepare a program that we will use in the course.<br><br>
 15. <a name="vs-code-settings"></a> Next we will configure VS Code.<br><br>
```
Comment (karlhorky) [inline linux.md:121]: looks like the numbering is off by 1 starting with 15

File: linux.md [hunk 4]
```diff
@@ -28,11 +28,23 @@ Make sure that you're running the 2nd-newest OS version or the newest version -
    corepack enable
    corepack prepare pnpm@latest --activate
    pnpm setup
+   source ~/`[[ $SHELL == *"zsh" ]] && echo '.zshrc' || echo '.bashrc'`
```
Comment (karlhorky) [inline linux.md:31]: @ProchaLu in [comment #2270597504](https://github.com/upleveled/system-setup/pull/91#issuecomment-2270597504)

> ## Linux
> I tested the System Setup Guide with @Eprince-hub [ubuntu-linux-github-actions](https://github.com/Eprince-hub/ubuntu-linux-github-actions) project. I had to install `pnpm` using this script
> 
> ```shell
> wget -qO- https://get.pnpm.io/install.sh | ENV="$HOME/.bashrc" SHELL="$(which bash)" bash -
> ```

this is not the same installation method we are using above

File: windows.md [hunk 5]
```diff
@@ -545,14 +545,14 @@ With those compatibility things out of the way, you're ready to start the system
...
-26. If you don't have one yet, create a GitHub account [here](https://github.com/join). Make sure to set a name.
+24. If you don't have one yet, create a Google account [here](https://accounts.google.com/signup?hl=en). Make a note of the email address associated with this account for usage in later steps.
+25. If you don't have one yet, create a GitHub account [here](https://github.com/join). Make sure to set a name.
 
     If you already have a GitHub account and you haven't set a name on GitHub yet, go to the [GitHub Profile Settings](https://github.com/settings/profile) and add a name:<br><br>
     <img src="./general-github-profile-settings.png"><br><br>
     We will use this name in the next step.<br><br>
 
-27. For this step, we'll need to **edit some of the information in the commands** by adding our own information.<br><br>
+26. For this step, we'll need to **edit some of the information in the commands** by adding our own information.<br><br>
```
Comment (karlhorky) [inline windows.md:555]: looks like the numbering is off, starting with 28 (outside of the diff, below)

File: windows.md [hunk 6]
```diff
@@ -253,10 +253,10 @@ With those compatibility things out of the way, you're ready to start the system
...
 
     ```bash
-    echo "export PATH=\$PATH:\"/c/Program Files/PostgreSQL/16/bin\"" >> "$USERPROFILE/.bash_profile"
-    echo "export PGDATA=\"/c/Program Files/PostgreSQL/16/data\"" >> "$USERPROFILE/.bash_profile"
-    echo "export PSQL_PAGER=\"less --chop-long-lines --header 1\"" >> "$USERPROFILE/.bash_profile"
-    source "$USERPROFILE/.bash_profile"
+    echo "export PATH=\$PATH:\"/c/Program Files/PostgreSQL/16/bin\"" >> ~/.bash_profile
+    echo "export PGDATA=\"/c/Program Files/PostgreSQL/16/data\"" >> ~/.bash_profile
+    echo "export PSQL_PAGER=\"less --chop-long-lines --header 1\"" >> ~/.bash_profile
+    source ~/.bash_profile
```
Comment (karlhorky) [inline windows.md:259]: maybe let's split this out to a separate PR, to avoid this getting lost in this PR

File: windows.md [hunk 7]
```diff
@@ -59,6 +59,7 @@ With those compatibility things out of the way, you're ready to start the system
    corepack enable
    corepack prepare pnpm@latest --activate
    pnpm setup
+   source "$USERPROFILE/.bash_profile"
```
Comment (karlhorky) [inline windows.md]: since [we'll switch from `$USERPROFILE` to `~`](https://github.com/upleveled/system-setup/pull/91#issuecomment-2264962880), we'll have a provlem if we keep this here - [it appears PowerShell doesn't support tilde](https://www.reddit.com/r/PowerShell/comments/14wdeih/tilde/)

we could move the Hyper install + configuration with Git Bash up earlier (eg. as a part of step 6 or a new step between step 6 and 7)

downside: the checking of the shell settings are further apart, and could become inconsistent.

Comment (karlhorky) [reply windows.md]: @ProchaLu decided to remove the `source` command from step 7 of the Windows guide in the latest commits

File: windows.md (hunk 1 above)
Comment (ProchaLu) [reply windows.md:57]: I checked it on Windows again, removed `nodejs-lts` from my system with all folders, also `pnpm`, and started new with the installation of `node`. For the installation, I used the same commands as the System Setup Guide as seen in the first screenshot. We can see the last command where I used `corepack enable`, and we get an error message that `corepack` is not recognized. 
corepack-missing

After restarting the terminal, `corepack` is recognized and I can use the latest `pnpm` version to install Preflight
corapack-restart

File: linux.md (hunk 2 above)
Comment (ProchaLu) [reply linux.md]: Added the new section for Windows, macOS and Linux in this commit
[ecb57fe](https://github.com/upleveled/system-setup/pull/91/commits/ecb57fe2e42067fd1de662f9002f6fcaf212d11c)

File: linux.md (hunk 3 above)
Comment (ProchaLu) [reply linux.md]: fixed numbers [ecb57fe](https://github.com/upleveled/system-setup/pull/91/commits/ecb57fe2e42067fd1de662f9002f6fcaf212d11c)

File: windows.md (hunk 5 above)
Comment (ProchaLu) [reply windows.md]: fixed numbers [ecb57fe](https://github.com/upleveled/system-setup/pull/91/commits/ecb57fe2e42067fd1de662f9002f6fcaf212d11c)

File: windows.md (hunk 6 above)
Comment (ProchaLu) [reply windows.md]: Created a new PR for these changes 
https://github.com/upleveled/system-setup/pull/94

File: linux.md (hunk 4 above)
Comment (ProchaLu) [reply linux.md:31]: I created a new Dockerfile that uses only a bash shell and installs the following software:

```
apt-get install -y build-essential git nodejs
```

I was able to recreate the installation steps for Upleveled Preflight. During testing, I found out that before installing Preflight, you need to source the shell configuration file to set up PNPM.


```bash
➜  ubuntu-linux-github-actions git:(main) ✗ docker run -it corepack-pnpm-test
root@3a10a6a19fb8:/# pnpm -v
bash: pnpm: command not found
root@3a10a6a19fb8:/# corepack enable
root@3a10a6a19fb8:/# corepack prepare pnpm@latest --activate
Preparing pnpm@latest for immediate activation...
root@3a10a6a19fb8:/# pnpm setup
Appended new lines to /root/.bashrc

Next configuration changes were made:
export PNPM_HOME="/root/.local/share/pnpm"
case ":$PATH:" in
  *":$PNPM_HOME:"*) ;;
  *) export PATH="$PNPM_HOME:$PATH" ;;
esac

To start using pnpm, run:
source /root/.bashrc
root@3a10a6a19fb8:/# pnpm add --global @upleveled/preflight
 ERR_PNPM_NO_GLOBAL_BIN_DIR  Unable to find the global bin directory

Run "pnpm setup" to create it automatically, or set the global-bin-dir setting, or the PNPM_HOME env variable. The global bin directory should be in the PATH.
root@3a10a6a19fb8:/# source ~/`[[ $SHELL == *"zsh" ]] && echo '.zshrc' || echo '.bashrc'`
root@3a10a6a19fb8:/# pnpm add --global @upleveled/preflight
Packages: +208
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Progress: resolved 208, reused 0, downloaded 208, added 208, done

/root/.local/share/pnpm/global/5:
+ @upleveled/preflight 7.0.9

Done in 2.2s
root@3a10a6a19fb8:/# 
```

File: windows.md [hunk 8]
```diff
@@ -92,33 +106,19 @@ With those compatibility things out of the way, you're ready to start the system
...
-    This will prepare a program that we will use in the course. This will take a while and then respond with a message with some usage instructions, similar to this:<br><br>
-    <img src="./general-cra-installed.avif"><br><br>
-
-14. Copy the following text and right-click in the PowerShell window to paste the text. Hit enter.<br><br>
-    ```bash
-    pnpm add --global @upleveled/preflight
-    ```
-    This will prepare a program that we will use in the course.<br><br>
-15. <a name="vs-code-settings"></a> Next we will configure VS Code.<br><br>
+13. Close PowerShell and open it again as administrator (like in step 2)<br><br>
```
[the remark this answers is not in the document]
Comment (ProchaLu) [reply windows.md]: This step is not necessary anymore. Removed and fixed the numbers in this commit [5324859](https://github.com/upleveled/system-setup/pull/91/commits/5324859e37acf59ceeb60c89e9b90795d80d96f0)

File: windows.md [hunk 9]
```diff
@@ -104,22 +149,7 @@ With those compatibility things out of the way, you're ready to start the system
     - [Refined GitHub Chrome Extension](https://chrome.google.com/webstore/detail/refined-github/hlepfoohegkhhmjieoechaddaejaokhf?hl=en)
     - [Socket Security Chrome Extension](https://chrome.google.com/webstore/detail/socket-security/jbcobpbfgkhmjfpjjepkcocalmpkiaop?hl=en)
     - [Web Vitals Chrome Extension](https://chrome.google.com/webstore/detail/web-vitals/ahfhijdlegdabablpippeagghigmibma?hl=en)
-12. Close PowerShell and open it again as administrator (like in step 2)<br><br>
```
Comment (ProchaLu) [inline windows.md:107]: Removed because it is unnecessary to restart PowerShell at this step

File: linux.md (hunk 2 above)
Comment (karlhorky) [reply linux.md]: the first line is not wrapped the same here, which leads to a scroll bar on a 13" screen:

Screenshot 2024-08-22 at 12 49 43

File: windows.md [hunk 10]
```diff
@@ -54,15 +54,60 @@ With those compatibility things out of the way, you're ready to start the system
...
+7. Close PowerShell and open it again as administrator (like in step 2). Copy the following text and right-click in the blue middle part of the PowerShell window to paste the text. Hit enter.<br><br>
    ```bash
    corepack enable
    corepack prepare pnpm@latest --activate
    Set-ExecutionPolicy -Scope Process -ExecutionPolicy Bypass
    pnpm setup
    ```
    This uses Corepack to install `pnpm`, and configures `pnpm`'s global bin directory.<br><br>
-8. Close PowerShell and open it again as administrator (like in step 2). Copy each line in the following text, right-click in the blue middle part of the PowerShell window to paste the text and hit enter.
+   Install `@upleveled/preflight`, a program we will use in the course, to verify that the previous commands were successful: copy the following text, paste it in the terminal and hit return.<br><br>
```
Comment (karlhorky) [inline windows.md:65]: seems like we should have empty lines around the paragraphs and code block, right?

File: windows.md [hunk 11]
```diff
@@ -54,15 +54,60 @@ With those compatibility things out of the way, you're ready to start the system
...
+
+   ```bash
+   pnpm add --global @upleveled/preflight
+   ```
+
+   Verify that your output looks similar to the "Successful" output below. If it looks very different (for example like the "Failing" output), repeat the pnpm installation commands at the start of this step.
```
Comment (karlhorky) [inline windows.md:71]: let's polish this (in all 3 guides)

```suggestion
   Verify that your output looks similar to the "Successful" output below. If it looks very different (for example like the "Failing" output), repeat the pnpm installation commands at the start of this step.
```

Comment (karlhorky) [inline windows.md:71]: let's polish this (in all 3 guides)

```suggestion
   Verify that your output looks similar to the "Successful" output below. If it looks very different (either like the "Failing" output or some other different output), repeat the pnpm installation commands at the start of this step.
```

Comment (karlhorky) [inline windows.md:71]: let's polish this (in all 3 guides)

```suggestion
   Verify that your output looks similar to the "Successful" output below. You can ignore differences in version numbers and package names - the important part is "Done in ..." on the last line.

   If it looks very different (either like the "Failing" output or like some other different output), repeat the pnpm installation commands at the start of this step.
```

Comment (karlhorky) [inline windows.md:71]: let's polish this (in all 3 guides)

```suggestion
   Verify that your output looks similar to the "Successful" output below. You can ignore any differences in version numbers and package names - the important part is "Done in ..." on the last line.

   If it looks very different (either like the "Failing" output or like some other different output), repeat the pnpm installation commands at the start of this step.
```

File: windows.md [hunk 12]
```diff
@@ -54,15 +54,60 @@ With those compatibility things out of the way, you're ready to start the system
...
    Set-ExecutionPolicy -Scope Process -ExecutionPolicy Bypass
    pnpm setup
    ```
    This uses Corepack to install `pnpm`, and configures `pnpm`'s global bin directory.<br><br>
-8. Close PowerShell and open it again as administrator (like in step 2). Copy each line in the following text, right-click in the blue middle part of the PowerShell window to paste the text and hit enter.
+   Install `@upleveled/preflight`, a program we will use in the course, to verify that the previous commands were successful: copy the following text, paste it in the terminal and hit return.<br><br>
+
+   ```bash
+   pnpm add --global @upleveled/preflight
+   ```
```
Comment (karlhorky) [inline windows.md:69]: it looks like this removal of the "Close PowerShell and open it again as administrator ..." may be introducing a new error:

in [this comment](https://github.com/upleveled/system-setup/issues/41#issuecomment-2082303483) it appears that `pnpm setup` does indeed set environment variables on the system and needs to have a new terminal started (see the bottom of the screenshot)

Screenshot 2024-04-29 104040

File: windows.md (hunk 1 above)
Comment (karlhorky) [reply windows.md:57]: removing the "Close PowerShell and open it again ..." from the following step may introduce a new error:

- https://github.com/upleveled/system-setup/pull/91#discussion_r1726895994

File: windows.md (hunk 10 above)
Comment (ProchaLu) [reply windows.md]: Added empty lines around the codeblock
[d295f33](https://github.com/upleveled/system-setup/pull/91/commits/d295f336fbe786aae30ba1358e97b15c8c00e0df)

File: linux.md (hunk 2 above)
Comment (karlhorky) [reply linux.md]: done in d295f336fbe786aae30ba1358e97b15c8c00e0df

**Review by karlhorky — APPROVED** [review]
LGTM (pending [the investigation into whether we can really remove the reopening after `pnpm setup`](https://github.com/upleveled/system-setup/pull/91#discussion_r1726895994) and re-adding of the step if necessary)

File: windows.md (hunk 12 above)
Comment (ProchaLu) [reply windows.md]: I successfully uninstalled Node and removed all the Windows environment variables used by pnpm. Reinstalled Node with `choco` and followed the System Setup steps.

In the screenshot, pnpm confirms that the setup is complete with the message 

```bash
Setup complete. Open a new terminal to start using pnpm.
```

pnpm-install
At the bottom of the screenshot, we can see that `pnpm` logs `Setup complete. Open a new terminal to start using pnpm`

I added the `Close PowerShell and open it again as administrator (like in step 2).` to the instructions in this commit
[43408f5](https://github.com/upleveled/system-setup/pull/91/commits/43408f5dcd8891b07e46794de0687a52d20ce962)
